Library/Homebrew/unpack_strategy.rb

```diff
@@ -96,7 +96,13 @@ def self.can_extract?(path:, magic_number:)
   private
 
   def extract_to_dir(unpack_dir, basename:)
-    FileUtils.cp_r File.join(path, "."), unpack_dir, preserve: true
+    directory_permissions = File.stat(unpack_dir).mode
+
+    begin
+      FileUtils.cp_r File.join(path, "."), unpack_dir, preserve: true
+    ensure
+      FileUtils.chmod directory_permissions, unpack_dir
```

Comment:
Can this fail on EPERM or similar?

Should this be running the chmod in the no-op case?